File: app/src/main/org/runnerup/core/util/RouteMarkerLabels.java
package org.runnerup.core.util;

import org.runnerup.data.entities.LocationEntity;

/** Lap marker info strings shared by map implementations. */
public final class RouteMarkerLabels {

  private RouteMarkerLabels() {}

  public static String lapMarkerInfo(
      Formatter formatter, LocationEntity loc, Double lapDistanceM) {
    long meters = lapDistanceM != null ? Math.round(lapDistanceM) : 0;
    return formatter.formatDistance(Formatter.Format.TXT_SHORT, meters)
        + " "
        + formatter.formatElapsedTime(
            Formatter.Format.TXT_SHORT, Math.round(loc.getElapsed() / 1000.0));
  }
}

File: app/src/main/kotlin/org/runnerup/core/util/RouteMarkerLabels.kt
package org.runnerup.core.util

import org.runnerup.data.entities.LocationEntity

/** Lap marker info strings shared by map implementations. */
object RouteMarkerLabels {
  @JvmStatic
  fun lapMarkerInfo(
      formatter: Formatter,
      loc: LocationEntity,
      lapDistanceM: Double?,
  ): String {
    val meters = if (lapDistanceM != null) kotlin.math.round(lapDistanceM).toLong() else 0L
    return formatter.formatDistance(Formatter.Format.TXT_SHORT, meters) +
        " " +
        formatter.formatElapsedTime(
            Formatter.Format.TXT_SHORT,
            kotlin.math.round(loc.elapsed / 1000.0).toLong(),
        )
  }
}
